.github/workflows/container-image.yml

```diff
@@ -0,0 +1,129 @@
+# The Build and Publish Container Image workflow builds container images and
+# pushes them to both GitHub Container Registry (GHCR) and Docker Hub.
+# It sets up QEMU and Docker Buildx for cross-platform builds,
+# and builds the container images using the Containerfile.
+# Upon pushes to the main branch or when releases are published,
+# it logs into GHCR and Docker Hub using credentials from GitHub secrets,
+# tags and pushes the images to both registries,
+# and generates and pushes signed build provenance attestations to each registry.
+# The workflow also triggers for pull requests to the main branch, verifying only the image build.
+
+name: Build and Publish Container Image
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  release:
+    types:
+      - published
+
+env:
+  # The variable ${{ github.repository }} is not suitable for container image names in our case because
+  # they must be lowercase, and our organization name is Icinga.
+  # Since our repository names are already lowercase, no additional modifications are necessary.
+  IMAGE_NAME: icinga/${{ github.event.repository.name }}
+
+jobs:
+  build-and-publish-container-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+      attestations: write
+      id-token: write
+
+    # TODO (el): This is a temporary setup; see the details in the "Build and push Container image" section.
+    strategy:
+      matrix:
+        platform: [linux/amd64, linux/arm64]
+
+    steps:
+      - name: Extract metadata (tags, labels)
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          # This will generate tags and labels for both the GHCR image and Docker Hub image.
+          images: |
+            # GitHub Container Registry
+            ghcr.io/${{ env.IMAGE_NAME }}
+            # Docker Hub
+            ${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.vendor=Icinga GmbH
+          tags: |
+            type=edge
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
```

Comment:
I am a bit unsure how the `latest` tag will be derived from these two inputs. Based on the [docs](https://github.com/docker/metadata-action?tab=readme-ov-file#latest-tag),  a `type=semver,pattern=...` will be used unless a `flavor` is defined.

So long we are not going to start supporting multiple release branches, everything should be fine.

Reply:
Good point. `latest` will be what is currently released, which in fact becomes a problem when we release from multiple branches.
I would like to make the workflow reusable, which is why I will fix it right away, since we already have repos that publish from multiple branches (support/*).
According to their docs, this should be correct then:

```
tags: |
  type=raw,value=latest,enable={{is_default_branch}}
  ...
```